lib/util/util.go

```diff
@@ -192,7 +193,7 @@ func (w NoopWriter) Write(bytes []byte) (int, error) {
 	return 0, nil
 }
 
-func GetRootDomains(domains []string) []string {
+func GetRootDomains(domains []string, blacklist bool) []string {
```

Comment:
Nit: maybe `pruneBlacklisted`, I read `blacklist` as an action when initially reading the signature.